src/zarr/core/array.py

```diff
@@ -2380,15 +2390,26 @@ def chunks_initialized(array: Array | AsyncArray) -> tuple[str, ...]:
 def _build_parents(node: AsyncArray | AsyncGroup) -> list[AsyncGroup]:
     from zarr.core.group import AsyncGroup, GroupMetadata
 
-    required_parts = node.store_path.path.split("/")[:-1]
-    parents = []
+    store = node.store_path.store
+    path = node.store_path.path
+    if not path:
```

Comment:
Do you know whether this branch is covered by an existing test? If not, it might be good to add one.